src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -277,7 +278,7 @@ public void UnregisterMessageTarget(ActivationData activation)
 
             if (activation.GrainInstance == null) return;
 
-            var grainTypeName = TypeUtils.GetFullName(activation.GrainInstanceType);
+            var grainTypeName = TypeUtils.GetFullName(activation.GrainInstanceType.GetTypeInfo());
```

Comment:
worth caching TypeInfo inside ActivatioData (init it when activationData.GrainInstanceType is inited) and reuse.